tensorflow/c/experimental/filesystem/filesystem_interface.h

```diff
@@ -83,6 +83,31 @@ typedef struct TF_TransactionToken {
   TF_Filesystem* owner;
 } TF_TransactionToken;
 
+typedef struct TF_Filesystem_Option_Value {
+  int type_tag;
+  int num_values;
+  union {
+    int64_t inv_val;
+    double real_val;
+    struct {
+      char* buf;
+      int buf_length;
+    } buffer_val;
+  } * values;  // owned
+  // A pointer to filesystem is needed in case options are passed around
+  // different modules to free the memory or allocate a new buffer for
+  // modification.
+  TF_Filesystem* file_system;
+} TF_Filesystem_Option_Value;
+
+typedef struct TF_Filesystem_Option {
+  char* name;                         // null terminated, owned
+  char* description;                  // null terminated, owned
+  int per_file;                       // bool actually, but bool is not a C type
+  TF_Filesystem_Option_Value* value;  // owned
+  TF_Filesystem* file_system;         // pointer to owning file system
```

Comment:
Is there some invariant relating `file_system` and `value->file_system`? Are they always the same?

Reply:
Since this part is only going to be touched through ModularFilesystem class, it may be better to track the filesystem there so I removed these.